internal/service/rpz/record_rpz_cname_ipaddress_data_source.go

```diff
@@ -0,0 +1,183 @@
+package rpz
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int32validator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	niosclient "github.com/infobloxopen/infoblox-nios-go-client/client"
+	"github.com/infobloxopen/infoblox-nios-go-client/rpz"
+	"github.com/infobloxopen/terraform-provider-nios/internal/flex"
+	"github.com/infobloxopen/terraform-provider-nios/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &RecordRpzCnameIpaddressDataSource{}
+
+func NewRecordRpzCnameIpaddressDataSource() datasource.DataSource {
+	return &RecordRpzCnameIpaddressDataSource{}
+}
+
+// RecordRpzCnameIpaddressDataSource defines the data source implementation.
+type RecordRpzCnameIpaddressDataSource struct {
+	client *niosclient.APIClient
+}
+
+func (d *RecordRpzCnameIpaddressDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "rpz_record_rpz_cname_ipaddress"
```

Comment:
```suggestion
	resp.TypeName = req.ProviderTypeName + "_" + "rpz_record_cname_ipaddress"
```

This change would have to be made for all examples and tests.